lib/core/assertion.js

```diff
@@ -84,12 +94,32 @@ module.exports = new (function() {
         passed = self.pass(value);
       }
 
-      var expected = typeof self.expected == 'function' ? self.expected() : self.expected;
-      self.client.assertion(passed, value, expected, self.message, self.abortOnFailure);
+      var timeSpent = new Date().getTime() - self.startTime;
+      if (!passed && timeSpent < self.timeout) {
+        return self._reschedule();
+      }
+
+      var expected = typeof self.expected == 'function' ? self.expected() : self.expected,
+          message = self._getFullMessage(passed, timeSpent);
```

Comment:
can we have a each variable on its own var statement? that's the current style throughout.